.github/workflows/nightly.yml

```
@@ -0,0 +1,88 @@
+name: 🌒 Nightly Release
+
+on:
+  schedule:
+    - cron: "0 7 * * *" # every day at 12AM PST
+
+jobs:
+  nightly:
+    name: 🌒 Nightly Release
+    if: github.repository == 'remix-run/remix'
+    runs-on: ubuntu-latest
+    outputs:
+      NEXT_VERSION: ${{ steps.version.outputs.NEXT_VERSION }}
+    steps:
+      - name: 🛑 Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: dev
+          token: ${{ secrets.FORMAT_PAT }}
+          fetch-depth: 0
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          cache: "yarn"
+
+      - name: 📥 Install deps
+        # even though this is called "npm-install" it does use yarn to install
+        # because we have a yarn.lock and caches efficiently.
+        uses: bahmutov/npm-install@v1
+
+      - name: 🏗 Build
+        run: yarn build
+
+      - name: ⤴️ Update Version
+        id: version
+        run: |
+          SHA=$(git rev-parse HEAD)
+          SHORT_SHA=${SHA::7}
+          NEXT_VERSION=0.0.0-nightly-${SHORT_SHA}
+          echo ::set-output name=NEXT_VERSION::${NEXT_VERSION}
+
+          git checkout -b nightly/${NEXT_VERSION}
+          npm version ${NEXT_VERSION}
+
+          git push origin --tags
+
+      - name: 🔐 Setup npm auth
+        run: |
+          echo "registry=https://registry.npmjs.org" >> ~/.npmrc
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> ~/.npmrc
+
+      - name: 🚀 Publish
+        run: npm run publish --tag nightly
+
+      - name: 🐱 Create GitHub release
+        uses: actions/create-release@v1
+        with:
+          draft: false
+          prerelease: true
+          release_name: v${{ steps.version.outputs.NEXT_VERSION }}
+          tag_name: v${{ steps.version.outputs.NEXT_VERSION }}
+        env:
+          # need this token in order to have it trigger the comment workflow
+          GITHUB_TOKEN: ${{ secrets.FORMAT_PAT }}
```

What is `FORMAT_PAT`? Should this be `secrets.GITHUB_TOKEN`? And is that comment right? Should this be:

```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

---

it needs to be _any_ github PAT other than the default GITHUB_TOKEN, otherwise the deployment tests and comment workflows wont run

---

updated to use `NIGHTLY_PAT` which i created from the remix-run-bot account :)